### PR TITLE
Modifed helper functions for Azure Cloud [PxB 2.7.1]

### PR DESCRIPTION
**Pull request by sabrarhussaini**
Modified the helper functions for azure backup location creation, azure bucket creation and azure bucket deletion to include the Azure China related changes as a part of Px-Backup 2.7.1.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #PB-6967

**Special notes for your reviewer**:
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/azurechina-byoc/98/
https://aetos.pwx.purestorage.com/resultSet/testSetID/655425

### CODE REVIEW

File: deployments/deploy-ssh.sh [hunk 1]
```diff
@@ -600,6 +600,8 @@ spec:
       value: "${TORPEDO_SSH_PASSWORD}"
     - name: TORPEDO_SSH_KEY
       value: "${TORPEDO_SSH_KEY}"
+    - name: AZURE_REGION
```
Comment (kshithijiyer-px) [inline deployments/deploy-ssh.sh:603]: Should this be Azure Endpoint?

File: tests/common.go [hunk 2]
```diff
@@ -5360,6 +5360,11 @@ func UpdateS3BackupLocation(name string, uid string, orgID string, cloudCred str
 func CreateAzureBackupLocation(name string, uid string, cloudCred string, cloudCredUID string, bucketName string, orgID string, validate bool) error {
 	backupDriver := Inst().Backup
 	encryptionKey := "torpedo"
+	azureRegion := os.Getenv("AZURE_REGION")
+	environmentType := api.S3Config_AzureEnvironmentType_AZURE_GLOBAL // Default value
+	if azureRegion == "CHINA" {
+		environmentType = api.S3Config_AzureEnvironmentType_AZURE_CHINA
+	}
```
[the remark this answers is not in the document]
Comment (sabrarhussaini) [reply tests/common.go:5367]: Sure @kphalgun-px , but this value will be passed from the pipeline as a choice parameter(CHINA/GLOBAL) so the region cannot be passed as lower case. But I can do this modification if needed, kindly advise

**Comment by mkoppal-px** [comment]
Why are the tests failing?

File: deployments/deploy-ssh.sh (hunk 1 above)
Comment (kshithijiyer-px) [reply deployments/deploy-ssh.sh:603]: How do you differentiate between this and bucket region?

File: tests/common.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kphalgun-px) [reply tests/common.go:5367]: Now, I think it is good to have, but it is not necessary, as it is very unlikely to run Torpedo on the cloud manually. Thanks.

File: deployments/deploy-ssh.sh (hunk 1 above)
Comment (sabrarhussaini) [reply deployments/deploy-ssh.sh:603]: @kshithijiyer-px Please confirm if I can modify this environment variable from AZURE_REGION to AZURE_STORAGE_ACCOUNT?

Comment (kshithijiyer-px) [reply deployments/deploy-ssh.sh:603]: @sabrarhussaini it will again be confusing let's make it `AZURE_ENDPOiNT`?

Comment (sabrarhussaini) [reply deployments/deploy-ssh.sh:603]: Sure @kshithijiyer-px

**Review by ak-px — APPROVED** [review]
LGTM, resolve conflicts

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**